scheds/rust/scx_rustlite/src/bpf/main.bpf.c

```diff
@@ -0,0 +1,370 @@
+/* Copyright (c) Andrea Righi <andrea.righi@canonical.com> */
+/*
+ * scx_rustlite: simple vtime-based scheduler written in Rust
+ *
+ *  The main goal of this scheduler is be an "easy to read" template that can
+ *  be used to quickly test more complex scheduling policies. For this reason
+ *  this scheduler is mostly focused on simplicity and code readability.
+ *
+ * The scheduler is made of a BPF part that implements the basic sched-ext
+ * functionalities and a user-space counterpart, written in Rust, that
+ * implement the scheduling policy itself.
+ *
+ * The default policy used by the user-space part is a based on virtual runtime
+ * (vtime):
+ *
+ * - each task receives the same time slice of execution
+ *
+ * - the actual execution time (adjusted based on the task's static priority)
+ *   determines the vtime
+ *
+ * - tasks are then dispatched from the lowest to the highest vtime
+ *
+ * The vtime is evaluated in the BPF part, tasks' then all the information are
```

Comment:
I had a bit of trouble parsing this sentence. Did it mean to say, "then all tasks' information are sent to the user-space part and stored in a red-black tree"? (applies to the rust comment header as well)

Reply:
I think I just fat fingered and cut some text, I'll rephrase it. :) but yeah the logic is pretty much what you said.

Reply:
I've rephrased like this, I think it's a bit more clear:
```
 * The BPF component evaluates the vtime for the tasks that are ready to run.
 * Subsequently, all task details are sent to the user-space component and
 * recorded in a red-black tree (indexed by vtime). The user-space part then
 * returns to the BPF component the list of tasks (ordered by their vtime) to
 * be dispatched following the order specified by the user-space.
```